eos-migrate-image-defaults.service

```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=Migrate eos-image-defaults to /var/lib
+ConditionPathIsDirectory=/var/eos-image-defaults
+ConditionPathIsSymbolicLink=!/var/eos-image-defaults
```

Comment:
I am thinking about how these conditions here interact with the non-atomicity of the operations you describe in the other comment.

From https://www.freedesktop.org/software/systemd/man/systemd.unit.html
```
If multiple conditions are specified, the unit will be executed if all of them apply (i.e. a logical AND is applied).
```

So the script will run whenever `/var/eos-image-defaults` is both a directory and not a symlink. If it gets interrupted after the first `mv`, the next boot it will not be a directory anymore, but it's not yet a symlink, leaving things in a broken state.

The documentation also says
```
Condition checks can be prefixed with a pipe symbol (|) in which case a condition becomes a triggering condition. If at least one triggering condition is defined for a unit, then the unit will be executed if at least one of the triggering conditions apply and all of the non-triggering conditions.
```

So perhaps we can simply make those triggering conditions instead.

That still won't solve the case when the script is interrupted after `ln` but before `sed`. For that reason, I believe that you should run `sed` at the beginning of the script instead (or at least before the last triggering condition).